unblob/handlers/archive/zip.py

```diff
@@ -137,6 +145,23 @@ def _parse_zip64(self, file: File, start_offset: int, offset: int) -> Instance:
             "Missing ZIP64 EOCD locator record header in ZIP chunk."
         )
 
+    def is_zip64(self, file, start_offset, offset, end_of_central_directory):
+        absolute_offset_of_cd = start_offset + end_of_central_directory.offset_of_cd
+
+        if 0 < absolute_offset_of_cd < offset:
+            file.seek(absolute_offset_of_cd, io.SEEK_SET)
+            try:
+                central_directory = self.cparser_le.cd_file_header_t(file)
+            except EOFError:
+                return False
```

Comment:
The source of this exception is an invalid header, that could occur for damaged zip files or for zip file in zip file.
The definition of `cd_file_header_t` has 2 variable length fields, that depend on size values embedded in the same record, these sizes could have damaged value leading to read past EOF.

As these fields are not used at all, and they do cost time to read, and could also fail, I propose to modify the definition and remove (comment out) the last 2 fields (`file_name`, `extra_field`).
Unfortunately removing these fields would lead to incomplete record reads, which cause problems when the zip cd is read through in a loop (`is_encrypted`), so that needs an extra seek adjustment to skip these 2 fields as well.
As an extra safety we could also check the parsed magic values for records and fail on mismatch instead of continuing with bad content.

----

I think we should also look at all the cd file headers to decide if we have zip64 format, e.g. we could have an archive with 2 files, the first one being a short one, while the second overflowing one of the 32bit size fields, making the archive zip64.
The current code would cut off the zip64 records at the end, leading to a potential extraction failure on the truncated chunk.